### PR TITLE
Support attachment base64 and custom filename.

### PR DESCRIPTION
**Pull request by zeetabit**
Support attachment base64 and custom filename regarding signal-cli update https://github.com/AsamK/signal-cli/pull/966.

Docker containers for checking:
https://hub.docker.com/repository/docker/bitzeta/signal-cli-rest-api

### CODE REVIEW

**Comment by bbernhard** [comment]
Many thanks for the contribution!

I just had a quick look at the changeset and at a first glance it looks really good (I'll do a bit of testing in the next days and then will merge the PR).

Just a small thing I've noticed: The `docs.go`, `swagger.json` and `swagger.yaml` files are auto generated by [swag](https://github.com/swaggo/swag). So, if you want to add something to the swagger documentation you would need to annotate it in the `api.go` sourcecode file and then run `swag init` inside the `src` folder. This then re-generates the files above.

File: src/client/client.go [hunk 1]
```diff
@@ -444,7 +461,7 @@ func (s *SignalClient) UnregisterNumber(number string, deleteAccount bool, delet
 		command := []string{"--config", s.signalCliConfig, "-a", number, "deleteLocalAccountData"}
 		_, err2 := s.cliClient.Execute(true, command, "")
 		if (err2 != nil) && (err != nil) {
-			err = fmt.Errorf("%w (%w)", err, err2)
+			err = fmt.Errorf("%w (%s)", err, err2.Error())
```
Comment (zeetabit) [inline src/client/client.go:464]: See error
```
fmt.Errorf call has more than one error-wrapping directive %w
```

**Comment by zeetabit** [comment]
Introduced tests with a couple of bug fixes. 
Added `go test` to run tests at docker build stage.

**Comment by zeetabit** [comment]
Introduced base64 file storing temporary to prevent limits in `signal-cli` native mode (works as before plus custom file name support).

**Comment by bbernhard** [comment]
Many thanks for this nice extension! Also thanks for taking the time to add some unittests - very much appreciated!

**Pull request merged**